onnx/external_data_helper.py

```diff
@@ -109,15 +111,17 @@ def convert_model_to_external_data(model, all_tensors_to_one_file=True, location
         if location:
             file_name = location
         for tensor in _get_all_tensors(model):
-            set_external_data(tensor, file_name)
+            if tensor.HasField("raw_data"):
+                set_external_data(tensor, file_name)
     else:
         for tensor in _get_all_tensors(model):
-            set_external_data(tensor, tensor.name)
+            if tensor.HasField("raw_data"):
+                set_external_data(tensor, tensor.name)
```

Comment:
Other than tensor without 'raw_data', do all other kinds of tensors need to be converted to external data? (The original method looks like brute-force to me and large model will be separated into a lot of files in this case) If yes, ORT would encounter some issues from some tensors with external data. https://github.com/onnx/onnx/issues/3157 Tensor checking and unpacking with external data needs to be fixed.

Once the decision is made, it would be great if ONNX can clarify the usage of external data here: https://github.com/onnx/onnx/blob/master/docs/IR.md#external-tensor-data.